include/util/exception.hpp

```diff
@@ -54,6 +56,219 @@ class exception final : public std::exception
     virtual void anchor() const;
     const std::string message;
 };
+
+/**
+ * Indicates a class of error that occurred that was caused by some kind of
+ * external input (common examples are out of disk space, file permission errors,
+ * user supplied bad data, etc).
+ */
+class runtime_error : public std::runtime_error
```

Comment:
Why do we inherit from `std::runtime_error` when we store a message and sourceref in this object in addition to the base's `what` string?